### PR TITLE
Updated JavaDoc for Graph class

### PR DESCRIPTION
**Pull request by cebe**
- added new and improved existing javadoc
- fixed some indentation and whitespace issues

You may review the changes by ignoring whitespace using the following URL:
https://github.com/project-flink/flink-graph/pull/54/files?w=0 (adding `?w=0` to ignore whitespace)

### CODE REVIEW

**Comment by ktzoumas** [comment]
Hey, looks good. Can you do a separate pull request with only the javadoc changes? That one should be merged

 The indentation will have to be fixed when the code moves to the Flink repo anyway, and part of this PR also contains new/deleted code.

**Comment by cebe** [comment]
Except the consistency about renaming the flink env to context in all places there are not code changes: https://github.com/project-flink/flink-graph/pull/54/files?w=0

I can remove the indentation though.

**Comment by cebe** [comment]
I have tried to revert most of the indentation changes but it is really hard to contribute when I need to try avoiding changes in indentation when writing code.

Can you please fix indentation on master yourself so that in future contributions it is easier for me to write code and for you to review the changes?

File: src/main/java/flink/graphs/Graph.java [hunk 1]
```diff
@@ -339,14 +374,14 @@ public void coGroup(Iterable<Edge<K, EV>> edges, Iterable<Tuple2<K, T>> input,
...
 	/**
-     * Apply value-based filtering functions to the graph 
-     * and return a sub-graph that satisfies the predicates
-     * for both vertex values and edge values.
-     * @param vertexFilter
-     * @param edgeFilter
-     * @return
-     */
-    public Graph<K, VV, EV> subgraph(FilterFunction<VV> vertexFilter, FilterFunction<EV> edgeFilter) {
+	 * Apply value-based filtering functions to the graph
```
Comment (vasia) [inline src/main/java/flink/graphs/Graph.java]: After your changes in #57 it's not value-based anymore :)

File: src/main/java/flink/graphs/Graph.java [hunk 2]
```diff
@@ -367,8 +402,8 @@ public void coGroup(Iterable<Edge<K, EV>> edges, Iterable<Tuple2<K, T>> input,
 	 * Apply value-based filtering functions to the graph
```
Comment (vasia) [inline src/main/java/flink/graphs/Graph.java]: same here

File: src/main/java/flink/graphs/Graph.java [hunk 3]
```diff
@@ -388,8 +423,8 @@ public void coGroup(Iterable<Edge<K, EV>> edges, Iterable<Tuple2<K, T>> input,
...
```
Comment (vasia) [inline src/main/java/flink/graphs/Graph.java]: and here :)

File: src/main/java/flink/graphs/Graph.java [hunk 4]
```diff
@@ -699,13 +734,14 @@ public void remove() {
 
 	/**
 	 * Creates a graph from a dataset of vertices and a dataset of edges
-	 * @param vertices
-	 * @param edges
-	 * @return
+	 * @param vertices a flink Dataset of vertices.
```
Comment (vasia) [inline src/main/java/flink/graphs/Graph.java]: I would prefer simply "DataSet" instead of "flink Dataset" (what other kind could it be?)

**Comment by vasia** [comment]
Hi @cebe!
Thanks a lot for this PR! I added some minor inline comments for you :)

File: src/main/java/flink/graphs/Graph.java (hunk 1 above)
Comment (cebe) [inline src/main/java/flink/graphs/Graph.java]: yeah right, will rebase it,

**Comment by cebe** [comment]
Done with the changes, ready to merge now?

**Comment by vasia** [comment]
Perfect! Thanks a lot.

**Pull request merged**